docs/monitoring.md

```diff
@@ -95,6 +95,44 @@ The history server can be configured as follows:
   </tr>
 </table>
 
+### Applying compaction of old event log files
+
+A long-running streaming application can bring a huge single event log file which may cost a lot to maintain and
+also requires a bunch of resource to replay per each update in Spark History Server.
+
+Enabling <code>spark.eventLog.rolling.enabled</code> and <code>spark.eventLog.rolling.maxFileSize</code> would
+let you have multiple event log files instead of single huge event log file which may help some scenarios on its own,
+but it still doesn't help you reducing the overall size of logs.
+
+Spark History Server can apply 'compaction' on the rolling event log files to reduce the overall size of
+logs, via setting the configuration <code>spark.history.fs.eventLog.rolling.maxFilesToRetain</code> on the
+Spark History Server.
+
+When the compaction happens, History Server lists all the available event log files, and considers the event log files older than
```

Comment:
"older than retained log files" .. This wording is confusing me a bit because you say "older".  Below the config is maxFilesToRetain, is not date based, its a number of files.  Am I missing something here or should the wording be somethign like like considers the event log files when there are more than spark.history.fs.eventLog.rolling.maxFilesToRetain?  Or is the intention that when there are more than spark.history.fs.eventLog.rolling.maxFilesToRetain, then it looks at the oldest of those log files?

Reply:
I chose "older" because the target event log files are created "before" than the event log files which will be retained - so that's semantically date based. It might be better to explain it with "index" which is clearer, but requires another explanation on "index".